### ISSUE TITLE
Notes from Nature Lab: update Talk links to Org-level

### ISSUE THREAD

**Issue by snblickhan** [zooniverse/front-end-monorepo#3574]
## Is your feature request related to a problem? Please describe.
NfN Labs (https://www.zooniverse.org/projects/md68135/notes-from-nature-labs) will become an FEM project when the next expedition launches, to beta test the DigiLeap OCR text-from-subject task. The Talk links on the project home page by default point to the project Talk board, but NfN prefers to use a single Talk board for all posts that aren't subject-specific (e.g. the Notes board or Done & Talk posts). 

## Describe the solution you'd like
I would like to redirect two of the Talk links on the project home page so that they point to the NfN team's requested Talk board (the 'main' NfN Talk board) instead of the NfN Labs Talk board. The URL for the 'main' NfN Talk board is: https://www.zooniverse.org/projects/zooniverse/notes-from-nature/talk.

See screenshots below for which links the team would like to change:
![Screen Shot 2022-08-26 at 4 41 31 PM](https://user-images.githubusercontent.com/25958358/186994589-a3abb5ec-b975-4c27-aa20-c2aa73b288d0.png)

![Screen Shot 2022-08-26 at 4 41 41 PM](https://user-images.githubusercontent.com/25958358/186994605-82e1da9d-6e19-4ae0-94f0-ecd27d238c44.png)

## Describe alternatives you've considered
None

## Additional context
The team is hoping to take this workflow to beta starting September 6th, so would be very helpful if we can update these links before that.

**Comment by eatyourgreens** [comment]
If it’s useful, we should try setting up Talk for their organisation, so that volunteers don’t have to be redirected to a different project. 

https://github.com/zooniverse/Panoptes-Front-End/pull/5415

**Comment by eatyourgreens** [comment]
Here’s a discussion for an organisation, from that PR. 
https://pr-5415.pfe-preview.zooniverse.org/organizations/eatyourgreens/my-first-organisation/talk/3164/2578806

**Comment by eatyourgreens** [comment]
In the code, you’ll want to override the project slug here to use a slug from a different project. 

https://github.com/zooniverse/front-end-monorepo/blob/5dc6506653e7ad6997d95945c5f9fd71e9ca3e3c/packages/app-project/src/screens/ProjectHomePage/components/MessageFromResearcher/MessageFromResearcherContainer.js#L53-L57

I’m not sure how you’d get that second project slug. Modify the project builder so that the team can add their own Talk link for a project? Hard code the NfN project into the FEM code (that seems like it would be hard to maintain)?

**Comment by eatyourgreens** [comment]
Here’s the code for the Join In link. Here you’ll need to override `owner` and `project` from the home page URL, and insert the NfN project name. 
https://github.com/zooniverse/front-end-monorepo/blob/5dc6506653e7ad6997d95945c5f9fd71e9ca3e3c/packages/app-project/src/screens/ProjectHomePage/components/ZooniverseTalk/components/JoinInButton/JoinInButtonContainer.js#L11-L15

**Comment by eatyourgreens** [comment]
Third option (might be more flexible): modify the Organisation lab so that organisation owners can add a Talk link for their organisation. Then modify project home pages to use the organisation Talk link when the project is part of an organisation.

**Comment by eatyourgreens** [comment]
Also noting that PFE projects within the NfN organisation don’t link to the NfN project Talk either eg. try the Join In link for Big Bee Bonanza!
https://www.zooniverse.org/projects/md68135/notes-from-nature-big-bee-bonanza

If we change this for one of their project home pages, we should change all their project home pages. Otherwise, we’re in the uncomfortable position where the same link behaves differently on different projects. Link behaviour should be consistent and predictable.

**Comment by eatyourgreens** [comment]
At the other end of the scale, the very quick solution would be to change the project name to `notes-from-nature` in Talk links for all NfN project home pages.

**Comment by mcbouslog** [comment]
Noting currently in PFE for any NfN project, like [Big Bee Bonanza](https://www.zooniverse.org/projects/md68135/notes-from-nature-big-bee-bonanza):
- the project home page "Join in" link goes to the project Talk (not the "organization" Talk)
- the project home page External Links section includes a link to the "organization" (actually old NfN project) Talk
- the project home page includes a link to the organization home page above the project description and in the project sub-navigation items (i.e. About, Classify, etc.)

Changing the noted FEM "Join in" and "Go to talk" links to go to the old NfN project Talk would have to be hardcoded specifically for NfN projects which I don't think is a good idea. Creating organization Talk pages should certainly happen, but I think will require more coordination with researchers using organizations and in NfN's use-case we might need to transfer the old NfN project Talk boards to the new organization Talk (if that's even possible) to avoid two different NfN general Talk pages?

I think if we want to get the NfN Labs project to beta soon it might be best to focus on getting organization links similar to PFE in the above/near the project description and in the project sub-nav and keep Talk links as is (which would match PFE)? In FEM we can add a link the "organization" Talk in the "Connect with us" section (which is same as the PFE External Links).

**Comment by mcbouslog** [comment]
I created #3579 (formerly just a card within the DigiLeap GitHub project) to track adding organization links to an FEM project home page.

**Comment by snblickhan** [comment]
Idea that came up when Mark and I were chatting about this today: rather than a separate, org-level board, what if we tried an approach that would allow a board (or boards) to appear on all projects within the organization? This could be set via a collaborator-level tick box when creating a board, indicating that board should appear on Talk for all projects within that org. That way, orgs can opt into this rather than having to use a structure that will be helpful to NfN but maybe isn't necessary for other orgs like Nest Quest Go, etc.

**Comment by eatyourgreens** [comment]
That’s what I had in mind here. Modify organisations so that the owner can specify an optional link, which would override the default Talk board on the home page. Maybe add some constraints so it can’t be abused for spam links. 

> Third option (might be more flexible): modify the Organisation lab so that organisation owners can add a Talk link for their organisation. Then modify project home pages to use the organisation Talk link when the project is part of an organisation.

**Comment by mcbouslog** [comment]
I really like the idea of an organization board(s) that would show on the projects' individual Talks! I'm not super familiar with Talk API, but I think it'd be fairly straightforward (maybe even no changes needed?). From the the frontend perspective, I think it'd also be fairly straightforward - on a project talk page, if the project is linked to an organization, then request organization boards, and show organization boards on the project talk page. I'm sure there are complications, but sounds good to me.

I think projects will always have to have the related subjects (Notes board) board, so this option brings the general/organization level boards to the projects.

**Comment by snblickhan** [comment]
> That’s what I had in mind here. Modify organisations so that the owner can specify an optional link, which would override the default Talk board on the home page. Maybe add some constraints so it can’t be abused for spam links.
> 
> > Third option (might be more flexible): modify the Organisation lab so that organisation owners can add a Talk link for their organisation. Then modify project home pages to use the organisation Talk link when the project is part of an organisation.

I think I confused the terminology here. This isn't really what I'm talking about -- I'm talking about an approach where you wouldn't even need a separate Talk page for an Org. In this line of thinking, we would allow teams to set specific _discussion_ board(s) to appear across all project Talk pages (nb by Board here in my original thread I was referring to a Discussion thread, sorry if my terminology confused things!).

So a team could have their own Talk page, with various discussion boards like Notes/Troubleshooting/Introduce Yourself/etc. but then there would also be a discussion board called "Org Updates" or whatever the team wanted to call it, which would also appear for every project in the Org. 

A benefit is that this wouldn't require volunteers to juggle separate Talk pages, as they do currently (i.e. going to the Project Talk for the Notes board, going to the Notes from Nature Talk for everything else).

**Comment by eatyourgreens** [comment]
Ah right. The Talk code, in PFE, doesn’t load organisations so, right now, Talk pages don’t have a concept of projects being part of an organisation.

**Comment by eatyourgreens** [comment]
You probably can’t easily load in a discussion board from another project, but you could use the PFE PR to create a Talk discussion board at the organisation level, then request that board from the project Talk of any project within the organisation. 

https://github.com/zooniverse/Panoptes-Front-End/pull/5415

**Comment by eatyourgreens** [comment]
Basically, create a board whose focus is the organisation ID, rather than the project ID. I think that might work, and it should be easy to get the organisation ID for a project.

**Comment by eatyourgreens** [comment]
The default section for a project Talk is set here, so maybe modify this to allow a project Talk to have more than one section? I’m not sure. 🤔
https://github.com/zooniverse/Panoptes-Front-End/blob/606680259004b988cb9599706f878794c57dc8cc/app/pages/project/talk.cjsx#L44

**Comment by eatyourgreens** [comment]
Permalinks for comments use `comment.project_slug`. That would break if a comment could be displayed across multiple projects, with different slugs. Something to watch out for.

**Comment by goplayoutside3** [comment]
@mcbouslog @snblickhan is this an Issue to keep open? I noticed it's sorta in the digileap "someday" project board, but wanted to double check.

**Comment by mcbouslog** [comment]
> Is this an Issue to keep open? I noticed it's sorta in the digileap "someday" project board, but wanted to double check.

Just my opinion - it is still an important topic, and the comments here are very helpful and still relevant, however, I don't think this GitHub issue is the best place for a discussion about organizations, talk, and how they're related/linked to continue. Though I'm not sure where the best place is for such a discussion (which could link to this maybe closed issue, as well as https://github.com/zooniverse/Panoptes-Front-End/pull/5415 , and maybe there's a few other related issues out there).

**Comment by snblickhan** [comment]
I agree with Mark -- there's definitely a much bigger Organizations discussion to be had. It was related to getting the Org level project link on FEM, which we did complete, so I think this is actually a good candidate to either stay open or get wrapped into a Org-centric discussion somewhere else.
